test/transmuxer.test.js

```diff
@@ -1265,49 +1310,346 @@ QUnit.test('infers sample durations from DTS values', function() {
 
   boxes = mp4.tools.inspect(segment);
   samples = boxes[0].boxes[1].boxes[2].samples;
-  QUnit.equal(samples.length, 3, 'generated three samples');
-  QUnit.equal(samples[0].duration, 1, 'set the first sample duration');
-  QUnit.equal(samples[1].duration, 2, 'set the second sample duration');
-  QUnit.equal(samples[2].duration, 2, 'inferred the final sample duration');
+  QUnit.equal(samples.length, 2, 'generated two samples, filters out pre-IDR');
+  QUnit.equal(samples[0].duration, 3, 'set the first sample duration');
+  QUnit.equal(samples[1].duration, 3, 'set the second sample duration');
```

Comment:
This duration is three because we re-use the last known duration for the final sample?